tractor/_debug.py

```diff
@@ -72,6 +78,18 @@ class Lock:
     # actor-wide variable pointing to current task name using debugger
     local_task_in_debug: Optional[str] = None
 
+    # NOTE: set by the current task waiting on the root tty lock from
+    # the CALLER side of the `lock_tty_for_child()` context entry-call
+    # and must be cancelled if this actor is cancelled via IPC
+    # request-message otherwise deadlocks with the parent actor may
+    # ensure
+    _debugger_request_cs: Optional[trio.CancelScope] = None
+
+    # NOTE: set only in the root actor for the **local** root spawned task
+    # which has acquired the lock (i.e. this is on the callee side of
+    # the `lock_tty_for_child()` context entry).
+    _root_local_task_cs_in_debug: Optional[trio.CancelScope] = None
```

Comment:
using this seems to cause a lot of test failures and *too-early-cancellation-of-the-child(s)* so keeping the code there but not currently using it inside the `._runtime` changes for now.